charms/jupyter-ui/tests/integration/test_charm.py

```diff
@@ -51,20 +56,56 @@ async def fetch_response(url):
     return result_status, str(result_text)
 
 
+async def get_unit_address(ops_test: OpsTest):
+    """Returns the unit address of jupyter-ui application."""
+    status = await ops_test.model.get_status()
+    jupyter_ui_units = status["applications"]["jupyter-ui"]["units"]
+    jupyter_ui_url = jupyter_ui_units["jupyter-ui/0"]["address"]
+    return jupyter_ui_url
+
+
 async def test_ui_is_accessible(ops_test: OpsTest):
     """Verify that UI is accessible."""
     # NOTE: This test is re-using deployment created in test_build_and_deploy()
     # NOTE: This test also tests Pebble checks since it uses the same URL.
-
-    status = await ops_test.model.get_status()
-    jupyter_ui_units = status["applications"]["jupyter-ui"]["units"]
-    jupyter_ui_url = jupyter_ui_units["jupyter-ui/0"]["address"]
+    jupyter_ui_url = await get_unit_address(ops_test)
 
     # obtain status and response text from Jupyter UI URL
-    port = CONFIG["options"]["port"]["default"]
-    result_status, result_text = await fetch_response(f"http://{jupyter_ui_url}:{port}")
+    result_status, result_text = await fetch_response(f"http://{jupyter_ui_url}:{PORT}")
 
     # verify that UI is accessible (NOTE: this also tests Pebble checks)
     assert result_status == 200
     assert len(result_text) > 0
     assert "Jupyter Management UI" in result_text
+
+
+async def test_notebook_image_selector(ops_test: OpsTest):
```

Comment:
Did you consider parametrizing this test?  It is a good test, but I think you could write it a bit cleaner using `@pytest.mark.parametrize` and passing one type of image at a time